app/services/fixity_checker.rb

```diff
@@ -59,16 +59,39 @@ def prune_checks
   # The current rules are:
   # Never throw out FAILED checks
   # Never throw out the earliest PASSED check
+  # Always keep a PASSED check if it is preceded or followed by a failing check.
   # Always keep N recent PASSED checks
   def checks_to_delete
-    # Throw all the passed checks INTO the trash.
     checks = FixityCheck.checks_for(@asset, permanent_url)
     return [] if checks.empty?
+
+    # Throw all the passed checks INTO the trash.
     trash = checks.select { | ch | ch.passed? }
+
     # But then, pop the earliest passed check back OUT of the trash.
     earliest_passed_check = trash.pop
-    # Finally, shift the most recent N passed checks back OUT of the trash.
+
+    # Shift the most recent N passed checks back OUT of the trash.
     recent_passed_checks = trash.shift(NUMBER_OF_RECENT_PASSED_CHECKS_TO_KEEP)
+
+    # One more thing:
+    # Certain passing checks are important for
+    # guessing the time window when something bad happened
+    # to a file, or when a bad file was fixed:
+    checks.each_with_index do | ch, i |
+      # Ignore this check if we weren't going to delete it:
+      next unless trash.include?(ch)
+      # OK, we're looking at a passed check we were
+      # about to delete.
+      if checks[i+1]&.failed? || checks[i-1]&.failed?
+        # This check passed, BUT it's either
+        # preceded or followed by a failing check.
+        # Let's remove it from the trash, so it
+        # won't get deleted.
+        trash.delete(ch)
+      end
```

Comment:
the implementation already in chf_sufia that doesn't rely on a 'trash' seems simpler to me. It just goes through the ordered log entries one by one, and decides whether to delete each one. What if I try putting that imp in here and see if specs still pass?

Reply:
Yeah, I did take a look. Only difference is NUMBER_OF_RECENT_PASSED_CHECKS_TO_KEEP is hardcoded in sufia, I believe.

Reply:
Hm, what  I meant is that the sufia logic doesn't use a `trash` array, does it? It has less steps, it doesn't put everything in a `trash` array first, it just goes through the things and deletes them immediately.

Reply:
I'm happy to do this, btw -- let me know. It shouldn't take long.

Reply:
By "only difference" I mean the only difference to pay attention to when porting the code.

Reply:
Yeah, really all I mean is "get rid of the temporary `trash` array" and do it more like chf_sufia in that regard. So if you wanna do that. 

but you can also say "Dude, it doesn't matter, this is micro-critiquing the code in a way that just doesn't matter", and I'll take that too.

Reply:
Nah, the Sufia way is simpler and better. Will refactor tomorrow; should be easy.

Reply:
Didn't get to it Friday, but done now.